src/WooCommerce.php

```diff
@@ -1158,4 +1174,15 @@ public static function getVariationIdByAttributes($product_id, array $attributes
     );
   }
 
+  /**
+   * Disables output of related products if over-ride checkbox is enabled.
+   */
+  public static function disableRelatedProducts(): void {
+    global $product;
+    $product_id = $product->get_id();
+    if (get_post_meta($product->get_id(), '_' . Plugin::PREFIX . '_disable_related_products', true) === 'yes') {
+      add_filter('woocommerce_product_related_posts_force_display', false, $product_id, 20, 2);
```

Comment:
If you look at the docs for [add_filter](https://developer.wordpress.org/reference/functions/add_filter/), it shows that the second argument needs to be a callable and third is priority. 

The second argument ` '__return_false'` is equivalent to `function() { return false; }` 

Perhaps this is working because $product_id is a really high number, so it's adding the filter so it executes last.

Do you think this will work?
```suggestion
      add_filter('woocommerce_product_related_posts_force_display',  '__return_false', 99999, 20, 2);
```

Reply:
Oh... I played around with priority as well by giving it a higher priority (e.g. priority 10). I did not consider it would need lower priority.

Reply:
Does it just ignore the last parameter? And wouldn't this make accepted args: 20? I'm confused.

Reply:
Oh yea my suggestion wasn't the best example. Would this work? 

```
add_filter('woocommerce_product_related_posts_force_display',  '__return_false', 99999, 2);
```

Reply:
Yes this works too so it's definitely the priority. Just need to put this on staging.